pandas/core/groupby/generic.py

```diff
@@ -1863,15 +1864,144 @@ def _transform_general(self, func, engine, engine_kwargs, *args, **kwargs):
     3  5  9
     4  5  8
     5  5  9
+
+    Using list-like arguments
+
+    >>> df = pd.DataFrame({"col": list("aab"), "val": range(3)})
+    >>> df.groupby("col").transform(["sum", "min"])
+        val
+       sum  min
+    0    1    0
+    1    1    0
+    2    2    2
+
+    .. versionchanged:: 3.0.0
+
+    Named aggregation
+
+    >>> df = pd.DataFrame({"A": list("aaabbbccc"), "B": range(9), "D": range(9, 18)})
+    >>> df.groupby("A").transform(
+    ...     b_min=pd.NamedAgg(column="B", aggfunc="min"),
+    ...     c_sum=pd.NamedAgg(column="D", aggfunc="sum")
+    ... )
+       b_min  c_sum
+    0      0     30
+    1      0     30
+    2      0     30
+    3      3     39
+    4      3     39
+    5      3     39
+    6      6     48
+    7      6     48
+    8      6     48
+
+    .. versionchanged:: 3.0.0
     """
     )
 
     @Substitution(klass="DataFrame", example=__examples_dataframe_doc)
     @Appender(_transform_template)
-    def transform(self, func, *args, engine=None, engine_kwargs=None, **kwargs):
-        return self._transform(
-            func, *args, engine=engine, engine_kwargs=engine_kwargs, **kwargs
+    def transform(
+        self,
+        func: None
+        | (Callable | str | list[Callable | str] | dict[str, NamedAgg]) = None,
+        *args,
+        engine: str | None = None,
+        engine_kwargs: dict | None = None,
+        **kwargs,
+    ) -> DataFrame:
+        if func is None:
+            # Convert named aggregations to dictionary format
+            transformed_func = self._named_agg_to_dict(*kwargs.items())
+            kwargs = {}
+            if isinstance(transformed_func, dict):
```

Comment:
Why is this `if` statement needed?

Reply:
I guess this check redundant, will remove it.